language/move-lang/src/naming/translate.rs

```diff
@@ -845,7 +845,11 @@ fn exp_(context: &mut Context, e: E::Exp) -> N::Exp {
             assert!(context.has_errors());
             NE::UnresolvedError
         }
-        EE::Index(..) | EE::Lambda(..) => panic!("unexpected specification construct"),
+        EE::Index(..) | EE::Lambda(..) |
+        // matches name variants only allowed in specs (we handle the allowed ones above)
+        EE::Name(..) => {
+            panic!("unexpected specification construct")
```

Comment:
Nit, slap an `ICE` on the front of that message.

Its a small thing, but I try to put it on all of the panics/asserts (at least I try to, I've probably missed some) so that its super clear this is some weird internal compiler failure 

At some point in the future I'd like to migrate these to a different `ice!(..)` macro that dumps a bunch more info

Reply:
Done